modules/programs/zsh.nix

```diff
@@ -188,6 +188,20 @@ let
     };
   };
 
+  historySubstringSearchModule = types.submodule {
+    options = {
+      enable = mkEnableOption "enable";
+      upAndDownKeys = mkOption {
+        default = [ "^[[A" "^[[B" ];
+        type = types.listOf types.str;
```

Comment:
Can you split this option?
Using a list is opaque for this sort of option and does not work well if you want to override only one keybind.
It also leads to weird logic like the `elemAt` in the `config` section.